lib/puppet/util/windows/taskscheduler2.rb

```diff
@@ -0,0 +1,707 @@
+#require 'puppet/util/windows/error'
+#require 'win32ole' # TODO should guard this on Windows only
+
+# The TaskScheduler2 class encapsulates taskscheduler settings and behavior using the v2 API
+# https://msdn.microsoft.com/en-us/library/windows/desktop/aa383600(v=vs.85).aspx
+
+# @api private
+class Puppet::Util::Windows::TaskScheduler2
+
+  # include Puppet::Util::Windows::String
+
+  # TODO May not need this if no COM
+  # require 'ffi'
+  # extend FFI::Library
+
+  # The error class raised if any task scheduler specific calls fail.
+  class Error < Puppet::Util::Windows::Error; end
+
+  # private
+
+  # # TODO needed?
+  # class << self
+  #   attr_accessor :com_initialized
+  # end
+
+  public
+
+  # The name of the root folder for tasks
+  ROOT_FOLDER = '\\'
+  
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/aa378137(v=vs.85).aspx
+  S_OK = 0
+
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/aa383558(v=vs.85).aspx
+  TASK_ENUM_HIDDEN  = 0x1
+
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/aa380596(v=vs.85).aspx
+  TASK_ACTION_EXEC = 0
+  TASK_ACTION_COM_HANDLER = 5
+  TASK_ACTION_SEND_EMAIL = 6
+  TASK_ACTION_SHOW_MESSAGE = 7
```

Comment:
We're only using the first constant... do we need the rest?

Reply:
As this class should be for any V2 API then I believe they should be kept

Reply:
I don't think we'll use any of them... not sure if they give us extra meta for enumerated tasks?  But I'm not even sure that they do! (Docs aren't clear / can't find examples)